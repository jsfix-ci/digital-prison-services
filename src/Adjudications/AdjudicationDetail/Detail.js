import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { H3 } from '@govuk-react/heading/lib/presets'
import { LabelAndValue, IncidentDetails } from './AdjudicationDetail.styles'

export const Detail = ({ detail }) => (
  <>
    <H3>Adjudication Details</H3>
    <GridRow>
      <GridCol setWidth="60%">
        <GridRow>
          <GridCol setWidth="one-third">
            <LabelAndValue label="Incident date and time" value={detail.incidentTime} />
          </GridCol>
          <GridCol setWidth="one-third">
            <LabelAndValue label="Type" value={detail.reportType} />
          </GridCol>
          <GridCol setWidth="one-third">
            <LabelAndValue label="Establishment" value={detail.establishment} />
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <IncidentDetails label="Detail" value={detail.incidentDetails} />
          </GridCol>
        </GridRow>
      </GridCol>
      <GridCol setWidth="40%">
        <GridRow>
          <GridCol setWidth="one-half">
            <LabelAndValue label="Location" value={detail.interiorLocation} />
          </GridCol>
          <GridCol setWidth="one-half">
            <LabelAndValue label="Report number" value={detail.reportNumber} />
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol setWidth="one-half">
            <LabelAndValue label="Reported by" value={detail.reporterName} />
          </GridCol>
          <GridCol setWidth="one-half">
            <LabelAndValue label="Report date and time" value={detail.reportTime} />
          </GridCol>
        </GridRow>
      </GridCol>
    </GridRow>
    <hr />
  </>
)

Detail.propTypes = {
  detail: PropTypes.shape({
    incidentTime: PropTypes.string,
    reportType: PropTypes.string,
    establishment: PropTypes.string,
    incidentDetails: PropTypes.string,
    interiorLocation: PropTypes.string,
    reportNumber: PropTypes.number,
    reporterName: PropTypes.string,
    reporterTime: PropTypes.string,
  }),
}

const mapStateToProps = ({ adjudicationHistory: { detail } }) => ({
  detail,
})

export default connect(mapStateToProps)(Detail)
