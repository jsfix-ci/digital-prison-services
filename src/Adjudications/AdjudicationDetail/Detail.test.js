import React from 'react'
import testRenderer from 'react-test-renderer'

import { Detail } from './Detail'

describe('Hearing', () => {
  it('should render empty data correctly', () => {
    const detail = {}

    const wrapper = testRenderer.create(<Detail detail={detail} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should render present detail correctly', () => {
    const detail = {
      incidentTime: '12/04/2014 - 12:32',
      reportType: 'General',
      establishment: 'Moorland HMP',
      incidentDetails: 'A crime occurred',
      interiorLocation: 'WING-A',
      reportNumber: 123456,
      reporterName: 'Jo Smith',
      reporterTime: '15/04/2014 - 10:00',
    }

    const wrapper = testRenderer.create(<Detail detail={detail} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
