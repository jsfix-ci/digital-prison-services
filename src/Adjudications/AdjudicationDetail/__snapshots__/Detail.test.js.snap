// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hearing should render empty data correctly 1`] = `
<div
  className="sc-EHOje jyflZT"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Report details
  </h3>
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH iPnSaj"
    >
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH LjyEw"
        >
          <strong
            className="label"
          >
            Incident date and time
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
        <div
          className="sc-bwzfXH LjyEw"
        >
          <strong
            className="label"
          >
            Type
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
        <div
          className="sc-bwzfXH LjyEw"
        >
          <strong
            className="label"
          >
            Location
          </strong>
          <p
            className="sc-ifAKCX isyHfT"
          >
            --
          </p>
          <p>
            --
          </p>
        </div>
      </div>
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH iXLolC"
        >
          <strong
            className="label"
          >
            Detail
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
      </div>
    </div>
    <div
      className="sc-bwzfXH jqxiAx"
    >
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH iIsRPd"
        >
          <strong
            className="label"
          >
            Report number
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
        <div
          className="sc-bwzfXH iIsRPd"
        >
          <strong
            className="label"
          >
            Reported by
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
      </div>
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH iIsRPd"
        >
          <strong
            className="label"
          >
            Report date and time
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hearing should render present detail correctly 1`] = `
<div
  className="sc-EHOje jyflZT"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Report details
  </h3>
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH iPnSaj"
    >
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH LjyEw"
        >
          <strong
            className="label"
          >
            Incident date and time
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            12/04/2014 - 12:32
          </p>
        </div>
        <div
          className="sc-bwzfXH LjyEw"
        >
          <strong
            className="label"
          >
            Type
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            General
          </p>
        </div>
        <div
          className="sc-bwzfXH LjyEw"
        >
          <strong
            className="label"
          >
            Location
          </strong>
          <p
            className="sc-ifAKCX isyHfT"
          >
            WING-A
          </p>
          <p>
            Moorland HMP
          </p>
        </div>
      </div>
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH iXLolC"
        >
          <strong
            className="label"
          >
            Detail
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            A crime occurred
          </p>
        </div>
      </div>
    </div>
    <div
      className="sc-bwzfXH jqxiAx"
    >
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH iIsRPd"
        >
          <strong
            className="label"
          >
            Report number
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            123456
          </p>
        </div>
        <div
          className="sc-bwzfXH iIsRPd"
        >
          <strong
            className="label"
          >
            Reported by
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            Jo Smith
          </p>
        </div>
      </div>
      <div
        className="sc-bdVaJa gkOpjT"
      >
        <div
          className="sc-bwzfXH iIsRPd"
        >
          <strong
            className="label"
          >
            Report date and time
          </strong>
          <p
            className="sc-bxivhb ddqgmM"
          >
            --
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
